serverlessworkflow/modules/ROOT/pages/cloud/build-workflow-image-with-quarkus-cli.adoc

```diff
@@ -1 +1,173 @@
+
+.Example on pushing the built Application to Quay.io
+[source,shell]
+----
+$ quarkus build -Dquarkus.container-image.build=true \
+    -Dquarkus.container-image.group=spolti \
+    -Dquarkus.container-image.name=serverless-workflow-greeting-quarkus \
+    -Dquarkus.container-image.tag=1.0 \
+    -Dquarkus.container-image.registry=quay.io \
+    -Dquarkus.container-image.push=true
+----
+
+The command above will result on the following Container image: `quay.io/spolti/serverless-workflow-greeting-quarkus:1.0`
+
+
+==== Building the Serverless Application with Native Image
+
+And what about native builds? When we talk about Serverless we do expect small startup footprint which can be better when using Native Builds. Quarkus CLI allows you to do it by passing the _--native_ flag:
+
+.Build Native Image
+[source,shell]
+----
+$ quarkus build --native -Dquarkus.container-image.build=true \
+    -Dquarkus.container-image.group=spolti \
+    -Dquarkus.container-image.name=serverless-workflow-greeting-quarkus \
+    -Dquarkus.container-image.tag=1.0-native \
+    -Dquarkus.container-image.registry=quay.io \
+    -Dquarkus.native.container-build=true
+----
+
+The command above will result on the following Container image: `quay.io/spolti/serverless-workflow-greeting-quarkus:1.0-native`
+
+If you don't have GraalVM installed, you can specify the _**quarkus.native.container-build=true**_ system properties. This property will create a Linux executable. Notice that, Native builds take longer than usual java builds.
+
+=== Testing the built Container Image
+
+If you do have Docker installed, the image will also be pushed to Docker's local registry:
+
+.JVM image
+[source,shell]
+----
+$ docker run -it -p 8080:8080 quay.io/spolti/serverless-workflow-greeting-quarkus:1.0
+...
+INFO  [io.quarkus] (main) serverless-workflow-greeting-quarkus 1.22.1.Final on JVM (powered by Quarkus 2.9.2.Final) started in 1.302s
+----
+
+.Native Image
+[source,shell]
+----
+$ docker run -it -p 8080:8080 quay.io/spolti/serverless-workflow-greeting-quarkus:1.0-native
+...
+INFO  [io.quarkus] (main) serverless-workflow-greeting-quarkus 1.22.1.Final native (powered by Quarkus 2.9.2.Final) started in 0.039s
+----
+
+Notice that the Native image startup is more than 1 second faster than the first one.
+Let's invoke the application:
+
+With the native image started, execute the following command:
+
+[source,shell]
+----
+$ curl -X POST -H 'Content-Type:application/json' -H 'Accept:application/json' -d '{"workflowdata" : {"name": "John", "language": "English"}}' http://localhost:8080/jsongreet
+----
+
+As output, the following response will be returned:
+[source,json]
+----
+{
+    "id": "2acf710d-7e4a-481d-925c-dfd85a369987",
+    "workflowdata": {
+        "name": "John",
+        "language": "English",
+        "greeting": "Hello from JSON Workflow, "
+    }
+}
+----
+
```

Comment:
Maybe we can add a section "What's Next" and add a few more links. I believe the next step will be deploying to Minikube (and also the k8s one you added).

Reply:
Added.
So far, minikube and kubernetes.